pkg/util/proxy.go

```diff
@@ -0,0 +1,139 @@
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"github.com/asaskevich/govalidator"
+	"github.com/pkg/errors"
+	"os"
+	"strings"
+)
+
+const OpenShiftRegistryIp = "172.30.1.1"
+
+var defaultNoProxies = []string{"localhost", "127.0.0.1", OpenShiftRegistryIp}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	err := validateProxyURL(httpProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validateProxyURL(httpsProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	np := []string{}
+	np = append(np, defaultNoProxies...)
+	if noProxy != "" {
+		np = append(np, strings.Split(noProxy, ",")...)
+	}
+	config := ProxyConfig{
+		httpProxy:  httpProxy,
+		httpsProxy: httpsProxy,
+		noProxy:    np,
+	}
+
+	return &config, nil
+}
+
+// ProxyConfig returns a the proxy configuration as a slice, one element for each of the potential settings HTTP_PROXY,
+// HTTPS_PROXY and NO_PROXY. If proxies are not enabled an empty slice is returned.
+func (p *ProxyConfig) ProxyConfig() []string {
+	config := []string{}
+
+	if !p.Enabled() {
+		return config
+	}
+
+	if p.httpProxy != "" {
+		config = append(config, fmt.Sprintf("HTTP_PROXY=%s", p.httpProxy))
+	}
+
+	if p.httpsProxy != "" {
+		config = append(config, fmt.Sprintf("HTTPS_PROXY=%s", p.httpsProxy))
+	}
+	config = append(config, fmt.Sprintf("NO_PROXY=%s", p.NoProxy()))
+
+	return config
+}
+
+// HttpProxy returns the configured value for the HTTP proxy. The empty string is returned in case HTTP proxy is not set.
+func (p *ProxyConfig) HttpProxy() string {
+	return p.httpProxy
+}
+
+// HttpsProxy returns the configured value for the HTTPS proxy. The empty string is returned in case HTTPS proxy is not set.
+func (p *ProxyConfig) HttpsProxy() string {
+	return p.httpsProxy
+}
+
+// NoProxy returns a comma separated list of hosts for which proxies should not be applied.
+func (p *ProxyConfig) NoProxy() string {
+	if p.Enabled() {
+		return strings.Join(p.noProxy, ",")
+	} else {
+		return ""
+	}
+}
+
+// AddNoProxy appends the specified host to the list of no proxied hosts.
+func (p *ProxyConfig) AddNoProxy(host string) {
+	p.noProxy = append(p.noProxy, host)
+}
+
+// Sets the current config as environment variables in the current process.
+func (p *ProxyConfig) SetToEnvironment() {
+	if !p.Enabled() {
+		return
+	}
+
+	if p.httpProxy != "" {
+		os.Setenv("HTTP_PROXY", p.httpProxy)
+	}
+	if p.httpsProxy != "" {
+		os.Setenv("HTTPS_PROXY", p.httpsProxy)
+	}
+	os.Setenv("NO_PROXY", p.NoProxy())
+}
+
+// Enabled returns true if at least one proxy (HTTP or HTTPS) is configured. Returns false otherwise.
+func (p *ProxyConfig) Enabled() bool {
```

Comment:
IMO, `Enabled` sounds like "Enable is alway true". Adding `Is` as prefix modifies that thinking (at least to me :smile: ).